packages/dcos-net/extra/dcos-net-setup.py

```diff
@@ -15,110 +15,142 @@
         return 0
 
-    networkd = b'systemd-networkd.service'
-    networkd_path = '/etc/systemd/network'
+    return result.returncode
+
+
+def is_unit_active(unit: str) -> bool:
+    result = run(['systemctl', 'is-active', unit],
+                 stdout=subprocess.PIPE)
+    if result.returncode == 0:
+        return True
+
+    return False
+
+
+def copy_file(src, dst) -> bool:
+    try:
+        if not filecmp.cmp(src, dst):
+            return bool(shutil.copyfile(src, dst))
+
+        return False
+    except FileNotFoundError:
+        return bool(shutil.copyfile(src, dst))
+
 
-    # Check if there is networkd
-    result = subprocess.run(['systemctl', 'list-unit-files', networkd],
-                            stdout=subprocess.PIPE)
+def add_config(unit: str, src: str, path: str) -> int:
+    # Check if the unit exists
+    result = check_for_unit(unit)
     if result.returncode != 0:
         return result.returncode
-    if networkd not in result.stdout:
-        return result.returncode
 
     # Copy the configuration
-    bname = os.path.basename(src)
-    dst = os.path.join(networkd_path, bname)
+    dst = os.path.join(path, os.path.basename(src))
 
     # Ensure the destination directory exists
-    os.makedirs(networkd_path, mode=0o755, exist_ok=True)
-    if not safe_filecmp(src, dst):
-        shutil.copyfile(src, dst)
+    os.makedirs(path, mode=0o755, exist_ok=True)
 
-    # Restart networkd only if it's active
-    result = subprocess.run(['systemctl', 'is-active', networkd],
-                            stdout=subprocess.PIPE)
-    if result.returncode != 0:
-        result.returncode = 0
-        return result.returncode
+    replaced = copy_file(src, dst)
 
-    # Restart networkd only if the configuration is updated
-    mtime = os.path.getmtime(dst)
-    result = subprocess.run(['systemctl', 'show', '--value',
-                             '--property', 'ActiveEnterTimestamp',
-                             networkd], stdout=subprocess.PIPE)
-    if result.returncode == 0:
-        active_enter_timestamp = result.stdout.strip().decode()
-        try:
-            started = datetime.datetime.strptime(
-                active_enter_timestamp,
-                '%a %Y-%m-%d %H:%M:%S %Z')
-            if started.timestamp() > mtime:
-                return result.returncode
-        except ValueError:
-            logging.warning('Unexpected ActiveEnterTimestamp value: "%s"',
-                            active_enter_timestamp)
-
-    # Restart networkd
-    return subprocess.run(['systemctl', 'restart', networkd]).returncode
-
-
-def safe_filecmp(src, dst):
-    try:
-        return filecmp.cmp(src, dst)
-    except FileNotFoundError:
-        return False
+    # Restart unit only if it's active
+    if not is_unit_active(unit):
+        return 0
+
+    # Restart the unit only if the configuration is updated
+    if replaced:
+        return run(['systemctl', 'restart', unit]).returncode
```

Comment:
:grey_question: This might be simpler as:
```
config_replaced = copy_file(src, dst)
# Restart the unit only if configuration changed and unit is active
if config_replaced and is_unit_active(unit):
    ...
```

Reply:
Much clearer and simpler, thank you!